engine/base.ftl

```diff
@@ -71,11 +71,20 @@
 [/#function]
 
 [#function intSquareRoot value ]
+    [#local lastPerfect = 1 ]
     [#list 1..100 as square]
-        [#if square * square == value ]
+        [#local multiple = square * square ]
+        [#if multiple == value ]
             [#return square ]
         [/#if]
+        [#if multiple?floor == multiple ]
```

Comment:
value is an integer so when will multiple?floor not equal multiple?

Am I correct in that the function will now return the square root of the smallest perfect square greater than or equal to the provided value?

Reply:
Good point, removed the floor check 

Yes the idea is to find the closest perfect square which can then be used for the smallest subnet mask that we need. I've renamed the function and added a comment